src/github/client/interceptors.ts

```diff
@@ -86,12 +85,11 @@ export const handleFailedRequest = (logger: Logger) =>
 			const status = response?.status;
 			const errorMessage = `Error executing Axios Request ` + error.message;
 
-			const rateLimitResetHeaderValue:string = response.headers?.["x-ratelimit-reset"];
 			const rateLimitRemainingHeaderValue:string = response.headers?.["x-ratelimit-remaining"];
 			if(status === 403 && rateLimitRemainingHeaderValue == "0") {
 				logger.warn({ err: error }, "Rate limiting error");
-				const rateLimitReset: number = parseInt(rateLimitResetHeaderValue) || Date.now() / 1000 + ONE_HOUR_IN_SECONDS;
-				return Promise.reject(new RateLimitingError(rateLimitReset, 0, error, status));
+				const rateLimitingError = rateLimitErrorFromResponse(response, error);
```

Comment:
I've extracted exception creation to a separate function rateLimitErrorFromResponse, because the same logic is used for graphql error handling

Reply:
Wouldn't it make more sense to keep it in here and use the same logic of this function for both REST and GraphQL calls?

Reply:
We can't unfortunately, because the responses for normal API and graphql are different. Normal API relies with 403 error, while GraphQL returns 200 and the error in the response body.

However, I can incapsulate some logic in RateLimitingError constructor to avoid code duplication